dp/cloud/python/magma/radio_controller/services/active_mode_controller/service.py

```diff
@@ -152,11 +148,43 @@ def _list_cbsds(session: Session) -> State:
             contains_eager(DBCbsd.grants).
             joinedload(DBGrant.state),
         ).
-        filter(*query_filter).
+        filter(_build_filter()).
         populate_existing()
     )
 
 
+def _build_filter():
+    multi_step = [
+        DBCbsd.fcc_id, DBCbsd.user_id, DBCbsd.number_of_ports,
+        DBCbsd.min_power, DBCbsd.max_power, DBCbsd.antenna_gain,
+    ]
+    single_step = [
+        DBCbsd.latitude_deg, DBCbsd.longitude_deg, DBCbsd.height_m,
+        DBCbsd.height_type, DBCbsd.indoor_deployment,
+    ]
+    return and_(
+        DBRequest.id == None,  # noqa: E711
+        or_(
+            or_(
+                DBCbsd.should_deregister == True,
+                DBCbsd.is_deleted == True,
+            ),
+            and_(
+                DBCbsd.single_step_enabled == False,
+                not_null(multi_step),
+            ),
+            and_(
+                DBCbsd.single_step_enabled == True,
+                not_null(multi_step + single_step),
+            ),
+        ),
+    )
+
+
+def not_null(fields: List[Any]):
+    return and_(*[field != None for field in fields])  # noqa: E711
```

Comment:
some fields that are required for type B radios are optional for type A

Reply:
as discussed, let's leave it like this for now